loader/src/sources/heb/index.js

```diff
@@ -0,0 +1,177 @@
+
+// Maps H-E-B product names to our product names.
+const PRODUCT_NAMES = {
+  pfizer: VaccineProduct.pfizer,
+  moderna: VaccineProduct.moderna,
+  janssen: VaccineProduct.janssen,
+  pediatric_pfizer: VaccineProduct.pfizerAge5_11,
+};
+
+async function fetchRawData() {
+  const response = await httpClient(API_URL, {
+    // Bust caches with a random querystring.
+    searchParams: { v: Math.random() * 999999999999 },
+    responseType: "json",
+    timeout: 30000,
+  });
+
+  const lastModified = response.headers["last-modified"];
+
+  return {
+    validAt: lastModified
+      ? DateTime.fromHTTP(lastModified, { zone: "utc" }).toISO()
+      : undefined,
+    data: response.body,
+  };
+}
+
+/**
+ * H-E-B has several locations missing a store number.
+ * Not including those locations in results, or locations
+ * with no open appointment slots.
+ */
+async function getData(states) {
+  const { validAt, data } = await fetchRawData();
+  const checkedAt = new Date().toISOString();
+
+  return data.locations
+    .filter((location) => Boolean(location.storeNumber))
+    .filter((location) => location.openAppointmentSlots > 0)
+    .map((entry) => {
+      let formatted;
+      Sentry.withScope((scope) => {
+        scope.setContext("location", {
+          id: entry.storeNumber,
+          provider: "heb",
+        });
+        try {
+          formatted = formatLocation(entry, validAt, checkedAt);
+        } catch (error) {
+          warn(error);
+        }
+      });
+      return formatted;
+    })
+    .filter((location) => states.includes(location.state));
+}
+
+function warn(message, context) {
+  console.warn(`H-E-B: ${message}`, context);
+  // Sentry does better fingerprinting with an actual exception object.
+  if (message instanceof Error) {
+    Sentry.captureException(message, { level: Sentry.Severity.Info });
+  } else {
+    Sentry.captureMessage(message, Sentry.Severity.Info);
+  }
+}
+
+function formatAvailability(openSlots) {
+  if (openSlots > 0) {
+    return Available.yes;
+  } else {
+    return Available.no;
+  }
+}
+
+/**
+ * H-E-B includes "other" products in their list
+ * of available vaccine types. Likely a flu vaccine /
+ * not relevant for covid vaccine scheduling.
+ */
+function formatAvailableProducts(raw) {
+  if (!raw) return undefined;
+
+  return raw
+    .map((value) => {
+      const productType = value.manufacturer.toLowerCase();
+      const formatted = PRODUCT_NAMES[productType];
+      if (productType != "other" && !formatted) {
+        warn(`Unknown product type`, value.manufacturer);
+      }
+      if (value.openAppointmentSlots > 0) {
+        return formatted;
+      }
+    })
+    .filter(Boolean);
+}
+
+function formatLocation(data, checkedAt, validAt) {
+  if (!checkedAt) checkedAt = new Date().toISOString();
+
+  const external_ids = [
+    ["heb", `${data.storeNumber}`],
+    ["heb", `${LocationType.pharmacy}-${data.zip}-${data.street}`],
```

Comment:
So the main point of the external IDs is that they allow us to match up reports about the same locations from different systems (e.g. this and VaccineSpotter) so that we don’t create duplicates. But that means we need to come up with matching identifiers. This is slightly different than VaccineSpotter, which will cause things not to match:
- Where we have `LocationType.pharmacy` here, VS is using `data.type`.
- We are using `data.zip` as is, but VS takes only the first 5 characters.
- VS is using the [slug package](https://www.npmjs.com/package/slug), which at the very least downcases things and replaces spaces with `-`. Probably some other stuff, too.

For example, this location is `offsite-78363-700-university-blvd`, but the code here would generate `PHARMACY-78363-700 University Blvd`: http://getmyvax.org/api/edge/locations/fe9f0d7f-3a28-4871-b121-561a020b1db4

That said, this is getting complicated. A quick look shows there’s only one location right now with no store number that we've seen before and would need to match up, so it’s probably not the end of the world if you just drop this instead. :\

Reply:
:+1: I'll go ahead and drop this. Agreed it seems low value especially since HEB locations with no store numbers seem to also have no vaccine appointment eligibility in all but one instance.